File: PMUG02_16_2021_Project/End_Java/app/src/main/java/com/neudesic/pmugfebappend/database/PersonDao.java
package com.neudesic.pmugfebappend.database;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import com.neudesic.pmugfebappend.model.Person;

import java.util.List;

@Dao
public interface PersonDao {

    @Query("SELECT * FROM person_table")
    LiveData<List<Person>> getAllPerson();

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    void insert(Person person);
}

File: PMUG02_16_2021_Project/End_Kotlin/app/src/main/java/com/neudesic/pmug02_16_2021_kotlin/database/PersonDao.kt
package com.neudesic.pmug02_16_2021_kotlin.database

import androidx.lifecycle.LiveData
import androidx.room.*
import com.neudesic.pmug02_16_2021_kotlin.model.Person

@Dao
interface PersonDao {
    @Query("SELECT * FROM person_table")
    fun getAllPerson(): LiveData<List<Person>>

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    suspend fun insert(person: Person)

}